src/core/stac/qgsstacparser.cpp

```diff
@@ -60,43 +65,50 @@ QString QgsStacParser::error() const
 }
 
 QgsStacCatalog *QgsStacParser::catalog()
+{
+  return parseCatalog( mData );
+}
+
+QgsStacCatalog *QgsStacParser::parseCatalog( const nlohmann::json &data )
 {
   try
   {
-    const QString ver( QString::fromStdString( mData.at( "stac_version" ) ) );
+    const QString ver( QString::fromStdString( data.at( "stac_version" ) ) );
     if ( !isSupportedStacVersion( ver ) )
     {
       mError = QStringLiteral( "Unsupported STAC version: %1" ).arg( ver );
       return nullptr;
     }
 
-    const QString id( QString::fromStdString( mData.at( "id" ) ) );
-    const QString description( QString::fromStdString( mData.at( "description" ) ) );
+    const QString id( QString::fromStdString( data.at( "id" ) ) );
+    const QString description( getString( data.at( "description" ) ) );
```

Comment:
I see that in some cases we're now using getString(), in others QString::fromStdString() - what is the pattern when to use one or the other?

Reply:
I've noticed that some stac servers return `null` for some attributes that the spec explicitly states are type `string`.
I've added this `getString()` method that treats `null`s as empty strings and use it for non-crucial fields, like _license_, _title_, _details_ etc, so that parsing those catalogs does not fail, but refrained from using it for more important bits, like `id`.